src/isar/state_machine/states/resuming_return_home.py

```diff
@@ -0,0 +1,64 @@
+from typing import TYPE_CHECKING, Callable, List, Optional
+
+from isar.apis.models.models import ControlMissionResponse
+from isar.eventhandlers.eventhandler import EventHandlerBase, EventHandlerMapping
+from isar.models.events import Event
+from robot_interface.models.exceptions.robot_exceptions import ErrorMessage
+
+if TYPE_CHECKING:
+    from isar.state_machine.state_machine import StateMachine
+
+
+class ResumingReturnHome(EventHandlerBase):
+
+    def __init__(self, state_machine: "StateMachine"):
+        events = state_machine.events
+
+        def _failed_resume_event_handler(
+            event: Event[ErrorMessage],
+        ) -> Optional[Callable]:
+            error_message: Optional[ErrorMessage] = event.consume_event()
+
+            if error_message is None:
+                return None
+
+            state_machine.events.api_requests.resume_mission.response.trigger_event(
+                ControlMissionResponse(
+                    success=False,
+                    failure_reason=(
+                        getattr(error_message, "error_reason", str(error_message))
+                        if hasattr(error_message, "error_reason")
+                        else str(error_message)
```

Comment:
We should make error_reason not be optional, but we can leave that for another PR